### ISSUE TITLE
Creating a list with depth great than 0 outputs incorrect AST for all nodes that follow

### ISSUE THREAD

**Issue by corbanbrook** [icelab/draft-js-ast-exporter#3]
1) Turn on list block mode
2) add an item
3) hit tab and add another item
4) end the list 
5) create a couple unstyled text blocks

The outputted ast will include the unstyled text blocks as children of the unordered-list-item block

**Comment by corbanbrook** [comment]
```
[
  [
    "block",
    [
      "unordered-list-item",
      "419f3",
      [
        [
          "inline",
          [
            [],
            "level 0"
          ]
        ],
        [
          "block",
          [
            "unordered-list-item",
            "9121g",
            [
              [
                "inline",
                [
                  [],
                  "level 1"
                ]
              ]
            ]
          ]
        ],
        [
          "block",
          [
            "unstyled",
            "8soca",
            [
              [
                "inline",
                [
                  [],
                  "level 0"
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]
```

**Comment by corbanbrook** [comment]
Here is the raw content state from draft js which shows that the third block has a depth of 0

```
{
  "entityMap": {},
  "blocks": [
    {
      "key": "419f3",
      "text": "level 0",
      "type": "unordered-list-item",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    },
    {
      "key": "9121g",
      "text": "level 1",
      "type": "unordered-list-item",
      "depth": 1,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    },
    {
      "key": "8soca",
      "text": "level 0",
      "type": "unstyled",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    }
  ]
}
```

**Comment by corbanbrook** [comment]
```
    } else if (lastBlock && block.getDepth() < lastBlock.getDepth() && parents.length > 0) {
      // This block is shallower, traverse up the set of parents
      let parent = parents.pop() // (mutating)
      currentContext = parent[dataSchema.block.children]
}
```

Looks like the issue is here. It just checks to see if the depth is less than the current depth and then pops the parent, which in this case would be the first list item. 

Rather than using the last blocks parent, it would need to find the context for the proper depth.

**Comment by makenosound** [comment]
@corbanbrook Ta for the report, and particularly for passing on your raw content state. You’re right about where the issue is — will see if I can sort out sometime today.

**Comment by makenosound** [comment]
@corbanbrook I’ve pushed a branch that should resolve this: https://github.com/icelab/draft-js-ast-exporter/pull/4. Any chance you could test with your setup to confirm it’s working as you’d expect?

**Comment by corbanbrook** [comment]
I'll be able to test this in a couple hours when I get into work. Thanks for the quick turn around time.

**Comment by makenosound** [comment]
Fixed in https://github.com/icelab/draft-js-ast-exporter/pull/4 and released as `v0.0.3`.
